pkg/openfeature/openfeature_test.go

```diff
@@ -329,6 +330,122 @@ func TestRequirement_1_6_1(t *testing.T) {
 	}
 }
 
+func TestRequirement_EventCompliance(t *testing.T) {
```

Comment:
The `event_executor` test suite covers a lot, but I think we could also use a couple "integration" (not unit) tests here as well that actually run handlers and test some high level behavior.

A couple scenarios I can think of:

- client handler fires when provider is set after client is created.
- client handler fires when provider is set before client is created.
- named client (bound to default provider) handler fires when the default provider emits an event
- named client (bound to named provider) handler fires when the default provider emits an event

Reply:
Good suggestion. Some of these are spec requirements this implementation is missing. I am working to implement them.

Reply:
All of these are covered. But I think I will add more. Eventing has lots of corner cases and we need lots of tests to validate them.